fedimint-client/src/lib.rs

```diff
@@ -1489,9 +1489,9 @@ impl Client {
         let mut recovery_receiver = self.client_recovery_progress_receiver.clone();
         recovery_receiver
             .wait_for(|in_progress| {
-                !in_progress
+                in_progress
                     .iter()
-                    .any(|(_id, progress)| !progress.is_done())
+                    .all(|(_id, progress)| progress.is_done())
```

Comment:
This only improves readability, there is no functional change here.

Reply:
>There's only so many combinations, we'll get it right eventually.

lol we are repeating combinations now

Reply:
Me and Manmeet playing with `!`, `any` and `all`.

![image](https://github.com/fedimint/fedimint/assets/9209/edf5883f-32a3-4adf-8913-ef2ae7283965)

Reply:
lmaoooo